jupyterlab_git/git.py

```diff
@@ -16,28 +17,70 @@ def __init__(self, root_dir, *args, **kwargs):
         super(Git, self).__init__(*args, **kwargs)
         self.root_dir = os.path.realpath(os.path.expanduser(root_dir))
 
-    def clone(self, current_path, repo_url):
+    def git_auth_input_wrapper(self, command, cwd, username, password):
+        p = pexpect.spawn(
+            command, 
+            cwd = cwd,
+        )
+        index = p.expect(['Username for .*: ', pexpect.EOF, pexpect.TIMEOUT])
+        if index==0:
```

Comment:
It is possible to remove `pexpect.EOF` and `pexpect.TIMEOUT` from the list of expected commands and handle them as exceptions, however down below in
```
p.expect([pexpect.EOF, 'fatal*', pexpect.TIMEOUT])
```
I cannot simply remove 'fatal*' as an expected option. `p.expect()` will hang on that instead of throwing exception.

Reply:
Ok, I've changed the pexpect wrapper, so that is more linear. All pexpect errors are handled using `except` with error code -1 and error stack message. All git CLI-related errors are handled by returning error code (using `p.wait()`) and error message (using `p.before()` -- i.e. returning all output between sending password and reaching EOF) directly from CLI.